dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/ContentletFactory.java

```diff
@@ -148,6 +148,19 @@ protected abstract Contentlet findContentletByIdentifierAnyLanguage(String ident
 	protected abstract Contentlet findContentletByIdentifier(final String identifier,
 			final Boolean live, final Long languageId, final String variantId) throws DotDataException, DotSecurityException;
 
+
+	/**
+	 * Retrieves a contentlet from the database based on its identifier and the working version
```

Comment:
probably the doc isnt accurate about the time machine nature

Reply:
true, fixed